lms/product/plugin/grouping_service.py

```diff
@@ -24,44 +24,84 @@ class GroupingServicePlugin:  # pragma: nocover
     sections_type: Optional[Grouping.Type] = None
     """The type of sections this plugin supports. `None` disables support."""
 
-    def get_sections_for_learner(self, svc, course) -> Optional[List]:
+    def get_sections_for_learner(
+        self, svc, course
+    ) -> Optional[List]:  # pragma: nocover
         """Get the sections from context when launched by a normal learner."""
 
         return None
 
-    def get_sections_for_instructor(self, svc, course: Course) -> Optional[List]:
+    def get_sections_for_instructor(
+        self, svc, course: Course
+    ) -> Optional[List]:  # pragma: nocover
         """Get the sections from context when launched by an instructor."""
 
         return None
 
     def get_sections_for_grading(
         self, svc, course: Course, grading_student_id
-    ) -> Optional[List]:
+    ) -> Optional[List]:  # pragma: nocover
         """Get the sections for a learner when they are being graded."""
 
         return None
 
     def get_groups_for_learner(
         self, svc, course: Course, group_set_id
-    ) -> Optional[List]:
+    ) -> Optional[List]:  # pragma: nocover
         """Get the sections from context when launched by a normal learner."""
 
         return None
 
     def get_groups_for_instructor(
         self, svc, course: Course, group_set_id
-    ) -> Optional[List]:
+    ) -> Optional[List]:  # pragma: nocover
         """Get the groups from context when launched by an instructor."""
 
         return None
 
     def get_groups_for_grading(
         self, svc, course: Course, group_set_id, grading_student_id
-    ) -> Optional[List]:
+    ) -> Optional[List]:  # pragma: nocover
         """Get the groups for a learner when they are being graded."""
 
         return None
 
+    def sections_enabled(self, request, application_instance, course) -> bool:
+        """Check if sections are enabled for this LMS, instance and course."""
+        return False  # Disabled by default
+
+    def group_set_id(self, request, assignment):
+        """
+        Get the group set ID for group launches.
+
+        A course can be divided in multiple "small groups" but it's possible to
+        have different sets of groups for the same course.
+
+        This ID identifies a collection of groups.
+        """
+        if not self.group_type or not assignment:
+            return None
+
+        return assignment.extra.get("group_set_id") if assignment else None
+
+    def launch_grouping_type(
+        self, request, application_instance, course, assignment
+    ) -> Grouping.Type:
```

Comment:
I think this is probably something which should be in the service. If we are adding plugin points (like `sections_enabled` and `group_set_id`) to customize the behavior, this looks like it should be the non-customizable part which lives in the service?

Also if we have the vague plan to do a plugin per grouping type (section / group etc) then this would be a method that lived over all of them. Which again would suggest the service.

Reply:
Yeah this makes a lot of sense in the service, moved there.

That simplified the testing as well.